testprog/testprog.go

```diff
@@ -9,8 +9,9 @@ import (
 )
 
 var (
-	completions      = []string{"bear\tan animal", "bearpaw\ta dessert", "dog", "unicorn\tmythical"}
-	specialCharComps = []string{"at@", "equal=", "slash/", "colon:", "period.", "comma,", "letter"}
+	completions                      = []string{"bear\tan animal", "bearpaw\ta dessert", "dog", "unicorn\tmythical"}
+	completionsWithSpecialCharacters = []string{"bash space", "bash\\escape", "bash\\ escaped\\ space", "bash>redirect", "bash#comment", "bash$var", "bash|pipe", "bash;semicolon"}
```

Comment:
This is hard to test by hand because I cannot get the shell to finish the completion since they all have the same `bash` string right before the special character.  Also, to avoid using double slashes, which I find confusing, can you use forward ticks as quotes?

So please change the above line to be:
```
completionsWithSpecialCharacters = []string{`bash1 space`, `bash2\escape`, `bash3\ escaped\ space`, `bash4>redirect`, `bash5#comment`, `bash6$var`, `bash7|pipe`, `bash8;semicolon`}
```
Then we can test by hand and do `testprog prefix special-chars bash1[tab][tab]`

Reply:
Done